### PR TITLE
fix: docstring for imagetorch encoder

### PR DESCRIPTION
**Pull request by rutujasurve94**
Docstring for `imagetorch` encoder

### CODE REVIEW

File: encoders/image/ImageTorchEncoder/README.md [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+- Run with Docker (`docker run`)
+  - ```bash
+    docker run jinahub/pod.encoder.imagetorchencoder:0.0.8-0.9.28 --port-in 55555 --port-out 55556
+    ```
+
+- Flow API
+  - ```python
+    from jina.flow import Flow
+    f = (Flow()
+        .add(name='my-encoder', image='jinahub/pod.encoder.imagetorchencoder:0.0.8-0.9.28', port_in=55555, port_out=55556)
```
Comment (Yongxuanzhang) [inline encoders/image/ImageTorchEncoder/README.md:37]:
```suggestion
        .add(name='my_encoder', uses='jinahub/pod.encoder.imagetorchencoder:0.0.8-1.0.4', port_in=55555, port_out=55556)
```

File: encoders/image/ImageTorchEncoder/README.md [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+    ```
+
+- Jina CLI
+  - ```bash
+    jina pod --uses jinahub/pod.encoder.imagetorchencoder:0.0.8-0.9.28 --port-in 55555 --port-out 55556
```
Comment (Yongxuanzhang) [inline encoders/image/ImageTorchEncoder/README.md:42]:
```suggestion
    jina pod --uses docker://jinahub/pod.encoder.imagetorchencoder:0.0.8-1.0.4 --port-in 55555 --port-out 55556
```

File: encoders/image/ImageTorchEncoder/README.md [hunk 3]
```diff
@@ -1,3 +1,56 @@
...
+    ```
+
+- Conventional local usage with `uses` argument
+  - ```bash
+    jina pod --uses hub/example/config.yml --port-in 55555 --port-out 55556
+    ```
+
+- Docker command
+
+  - Specify the image name along with the version tag. The snippet below uses Jina version `0.9.20`
```
Comment (Yongxuanzhang) [inline encoders/image/ImageTorchEncoder/README.md:52]: Or make sure the version is consistent
```suggestion
  - Specify the image name along with the version tag. The snippet below uses Jina version `1.0.4`
```

File: encoders/image/ImageTorchEncoder/README.md [hunk 4]
```diff
@@ -1,3 +1,56 @@
...
+
+  - ```bash
+    docker pull jinahub/pod.encoder.imagetorchencoder:0.0.8-0.9.28
```
Comment (Yongxuanzhang) [inline encoders/image/ImageTorchEncoder/README.md:55]:
```suggestion
    docker pull jinahub/pod.encoder.imagetorchencoder:0.0.8-1.0.4
```

File: encoders/image/ImageTorchEncoder/README.md [hunk 5]
```diff
@@ -1,3 +1,56 @@
...
+
+Initialise ImageTorchEncoder:
+
+`ImageTorchEncoder(model_path='pretrained', channel_axis=1, metas=metas, model_name=MobileNetV2)`
+
+Users can use Pod images in several ways:
+
+- Run with Docker (`docker run`)
+  - ```bash
+    docker run jinahub/pod.encoder.imagetorchencoder:0.0.8-0.9.28 --port-in 55555 --port-out 55556
```
Comment (Yongxuanzhang) [inline encoders/image/ImageTorchEncoder/README.md:30]:
```suggestion
    docker run jinahub/pod.encoder.imagetorchencoder:0.0.8-1.0.4 --port-in 55555 --port-out 55556
```

**Review by Yongxuanzhang — COMMENTED** [review]
There seems no 0.0.8-0.9.28 version. I think we can either use 0.0.7-1.0.4, or 0.0.8-1.0.4, or like @cristianmtr did in his PR. https://github.com/jina-ai/jina-hub/pull/4238/files

File: encoders/image/ImageTorchEncoder/__init__.py [hunk 6]
```diff
@@ -74,6 +76,14 @@ def _get_pooling(self, feature_map: 'np.ndarray') -> 'np.ndarray':
     @batching
     @as_ndarray
     def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
+        """
+        Encode input data into `np.ndarray`.
+        :param data: Image to be encoded, expected a `np.ndarray` of BatchSize x (Channel x Height x Width).
```
Comment (nan-wang) [inline encoders/image/ImageTorchEncoder/__init__.py:81]:
```suggestion

        :param data: Image to be encoded, expected a `np.ndarray` of BatchSize x (Channel x Height x Width).
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: encoders/image/ImageTorchEncoder/__init__.py (hunk 6 above)
Comment (davidbp) [reply encoders/image/ImageTorchEncoder/__init__.py:81]: Image/Images to be encoded, expected a `np.ndarray` of shape  BatchSize x (Channel x Height x Width)

**Comment by rutujasurve94** [comment]
Thanks a lot for the review, I have made changes as per all the suggestions.

File: encoders/image/ImageTorchEncoder/README.md [hunk 7]
```diff
@@ -1,3 +1,56 @@
...
+- Run with Docker (`docker run`)
+  - ```bash
+    docker run jinahub/pod.encoder.imagetorchencoder:0.0.8-1.0.4 --port-in 55555 --port-out 55556
+    ```
+
+- Flow API
+  - ```python
+    from jina.flow import Flow
+    f = (Flow()
+        .add(name='my_encoder', uses='jinahub/pod.encoder.imagetorchencoder:0.0.8-1.0.4', port_in=55555, port_out=55556)
```
Comment (nan-wang) [inline encoders/image/ImageTorchEncoder/README.md:37]:
```suggestion
        .add(name='my_encoder', uses='docker://jinahub/pod.encoder.imagetorchencoder:0.0.8-1.0.4', port_in=55555, port_out=55556)
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: encoders/image/ImageTorchEncoder/README.md [hunk 8]
```diff
@@ -1,3 +1,56 @@
 # ImageTorchEncoder
 
-`ImageTorchEncoder` encodes data from a ndarray, potentially B x (Channel x Height x Width) into a 
\ No newline at end of file
+`ImageTorchEncoder` encodes data from a ndarray, potentially BatchSize x (Channel x Height x Width) into a ndarray of `BatchSize * d`. Internally, :class:`ImageTorchEncoder` wraps the models from `tensorflow.keras.applications`. https://keras.io/applications/
```
Comment (bwanglzu) [inline encoders/image/ImageTorchEncoder/README.md:3]: can you fix this in the attached image:

File: encoders/image/ImageTorchEncoder/README.md [hunk 9]
```diff
@@ -1,3 +1,56 @@
...
+Initialise this Executor specifying parameters i.e.:
+
+| `param_name`  | `param_remarks` |
+| ------------- | ------------- |
+| `model_path`  | the directory path of the model in the `SavedModel` format  |
+| `model_name`  | includes `resnet18`, `alexnet`, `squeezenet1_0`, `vgg16`, `densenet161`, `inception_v3` |
+| `channel_axis`| axis of the color channel  |
+| `pool_strategy` | strategy of pooling operation |
+
+The model is pre-trained on [MobileNetV2] data
```
Comment (bwanglzu) [inline encoders/image/ImageTorchEncoder/README.md:17]: missing link for `MobileNetV2`

**Review by bwanglzu — CHANGES_REQUESTED** [review]
Great work! minor comments added

File: encoders/image/ImageTorchEncoder/README.md (hunk 9 above)
Comment (JoanFM) [reply encoders/image/ImageTorchEncoder/README.md:17]: MobileNetV2 data makes no sense, the dataset is ImageNet. But anyway I think is not worth, I would say `model` is one of `resnet18`, `alexnet`, `squeezenet1_0`, `vgg16`, `densenet161`, `inception_v3`

**Review by JoanFM — DISMISSED** [review]

File: encoders/image/ImageTorchEncoder/README.md (hunk 9 above)
Comment (rutujasurve94) [reply encoders/image/ImageTorchEncoder/README.md]: I've updated the doc to reflect the right models.

**Comment by rutujasurve94** [comment]
@JoanFM @bwanglzu thanks for the re-review, the PR is up-to-date with the latest reviewed changes

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Review by nan-wang — APPROVED** [review]
LGTM👍
